presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -332,7 +345,51 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
             case GREATER_THAN_OR_EQUAL:
                 return planQuantifiedOrderable(subPlan, quantifiedComparison, correlationAllowed);
         }
-        throw new SemanticException(NOT_SUPPORTED, quantifiedComparison, "Quantified comparison with the given quantifier and comparison type is not supported");
+        // all cases are checked, so this exception should never be thrown
+        throw new IllegalStateException("Unexpected combination of quantifier and comparison type: " +
+                quantifiedComparison.getQuantifier() + ", " + quantifiedComparison.getComparisonType());
+    }
+
+    private PlanBuilder planQuantifiedEqualsAll(PlanBuilder subPlan, QuantifiedComparisonExpression quantifiedComparison, boolean correlationAllowed)
+    {
+        checkArgument(quantifiedComparison.getComparisonType() == EQUAL && quantifiedComparison.getQuantifier() == Quantifier.ALL);
+        // A = ALL B <=> min B = max B && A = min B
+        RelationPlan subqueryRelationPlan = createRelationPlan(quantifiedComparison.getSubquery());
+        PlanNode subqueryPlan = subqueryRelationPlan.getRoot();
+        List<Expression> outputColumnReference = ImmutableList.of(getOnlyElement(subqueryRelationPlan.getOutputSymbols()).toSymbolReference());
+        Type outputColumnType = getOnlyElement(subqueryRelationPlan.getDescriptor().getAllFields()).getType();
+        if (!outputColumnType.isOrderable()) {
+            throwNotSupportedException(quantifiedComparison, "Quantified comparison with given parameters on unorderable type");
```

Comment:
With the second form above, it shouldn't be that much more expensive. The Distinct-limit optimization will cause the subquery to bail out as soon as it sees two distinct values, and won't use more memory than necessary to store just the two values.

It's fine to leave it as a future enhancement, though. Please file an issue to track it.